blockstore/src/main/java/brooklyn/location/blockstore/azure/arm/AzureArmVolumeManager.java

```diff
@@ -269,36 +241,35 @@ private boolean resourceGroupContainsVm(AzureComputeApi api, String resourceGrou
         return vmApi.get(vmName) != null;
     }
 
-    private boolean waitForJobToComplete(final AzureComputeApi api, final URI uri, Duration waitTime) {
-        checkNotNull(uri, "uri must not be null");
-        final JobApi jobApi = api.getJobApi();
-        final AtomicReference<JobStatus> latest = new AtomicReference<JobStatus>();
-        
-        boolean done = Repeater.create("Waiting job for URI to completee: " + uri)
+    private Disk waitDiskToAppear(final DiskApi diskApi, final String diskName, Duration waitTime) {
+        checkNotNull(diskName, "diskName must not be null");
+        final AtomicReference<Disk> latest = new AtomicReference<>();
+
+        Repeater.create("Waiting Disk to complete: " + diskName)
                 .every(1, TimeUnit.SECONDS)
                 .limitTimeTo(waitTime)
                 .until(new Callable<Boolean>() {
                     @Override
                     public Boolean call() throws Exception {
-                        JobStatus currentJobStatus = jobApi.jobStatus(uri);
-                        latest.set(currentJobStatus);
-                        return currentJobStatus == ParseJobStatus.JobStatus.DONE;
+                        Disk disk = diskApi.get(diskName);
+                        latest.set(disk);
+                        return disk != null && "Succeeded".equals(disk.properties().provisioningState());
```

Comment:
In case the disk creation fails this will be stuck waiting for `waitTime`. To abort early you could wait for `!"Updating".equals(...)` with an additional `Succeeded` check after the repeater completes.